metrotranscom/doorway
# Create combined listings endpoint in backend/core

# Pull Request Template

## Issue Overview

This PR addresses #31 


## Description

This PR adds a new listings endpoint, `/listings/combined`, which combines all internal listings with a new table, `external_listings`, that will hold listings from the existing Bloom instance.  It also includes migrations to create the new table with any necessary indexes and a view to combine the two tables together.  Optimizations were made to the table and SELECT query that reduce query time by 60x compared to the original `/listings` endpoint.

Code changes were made with a focus on making them easy to remove in the near future when no longer needed.  New classes are placed in `src/listings/combined` and all changes are commented with `REMOVE_WHEN_EXTERNAL_NOT_NEEDED` to make them easy to find later on.

## How Can This Be Tested/Reviewed?

Verify manually by building via `yarn build`, running with `yarn start`, and going to `http://localhost:3100/listings/combined` in your browser.  Verify e2e tests by running `yarn test:e2e:local -t combined`.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
